### PR TITLE
Add new troubleshooting codeql

### LINKED ISSUE #15967: [User feedback] Add new troubleshooting section to CodeQL article
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

[Troubleshooting the CodeQL workflow](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow)

### What part(s) of the article would you like to see updated?

Question from user:

> "In my workload, analysis fails on generated code (which is thus known to be good). How can I suppress such false positives?"

Docs update needed:

[Troubleshooting the CodeQL workflow](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow)

- In the section "Reduce the amount of code being analyzed in a single workflow", convert the 2nd, 3rd and 4th paragraphs to a reusable.
- Add a new section called "Alerts found in generated code" use the new reusable.

For information about reusables, see [Reusables and variables](https://github.com/github/docs/blob/main/contributing/content-style-guide.md#reusables-and-variables) and [docs/data/reusables/README.md](https://github.com/github/docs/blob/main/data/reusables/README.md)

### Additional information

Showing the same content twice in one article may seem strange. However, this will make it more discoverable for users who want to exclude code from analysis for reasons apart from their build taking too long.

### PR DESCRIPTION
**Pull request by cmwilson21**
### Why:

Closes [15967](https://github.com/github/docs/issues/15967)

### What's being changed:
Changed the second, third, and fourth paragraphs in "Reduce the amount of code being analyzed in a single workflow" section to a reusable titled - "alerts-found-in-generated-code" and created a new section with that same name. The new section was placed after "[Lines of code scanned are lower than expected](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow#lines-of-code-scanned-are-lower-than-expected)" and before the section "[Extraction errors in the database](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow#extraction-errors-in-the-database)". I'm happy to move it to a different spot if it makes more sense somewhere else.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@cmwilson21 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: data/reusables/code-scanning/alerts-found-in-generated-code.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
+
+For languages like Go, JavaScript, Python, and TypeScript, that {% data variables.product.prodname_codeql %} analyzes without compiling the source code, you can specify additional configuration options to limit the amount of code to analyze. For more information, see "[Specifying directories to scan](/code-security/secure-coding/configuring-code-scanning#specifying-directories-to-scan)."
+
+If you split your analysis into multiple workflows as described above, we still recommend that you have at least one workflow which runs on a `schedule` which analyzes all of the code in your repository. Because {% data variables.product.prodname_codeql %} analyzes data flows between components, some complex security behaviors may only be detected on a complete build.
```
Comment (felicitymay) [inline data/reusables/code-scanning/alerts-found-in-generated-code.md:5]: I've only just noticed that this last paragraph refers to the original first paragraph (the one that's not included in this new reusable). 🙈 

I think it probably makes sense to move line 5 out of the reusable and back into its original location, after the reusable is called.

**Review by felicitymay — COMMENTED** [review]
Generally looks great - thanks 💖 

Reviewing your PR made me realize an oversight in my original suggestion here, but once that's resolve, this should be ready to merge 👍🏻

File: data/reusables/code-scanning/alerts-found-in-generated-code.md (hunk 1 above)
Comment (cmwilson21) [reply data/reusables/code-scanning/alerts-found-in-generated-code.md]: Got it! Pushed the new changes just now :)

**Review by felicitymay — APPROVED** [review]
Thanks Courtney, that looks great 💖 

🚀

**Pull request merged**